core/bin/via_server/src/node_builder.rs

```diff
@@ -103,6 +110,36 @@ impl ViaNodeBuilder {
         Ok(self)
     }
 
+    // QueryEthClientLayer is mock, it's not used in the current implementation
+    fn add_query_eth_client_layer(mut self) -> anyhow::Result<Self> {
+        let genesis = self.genesis_config.clone();
+        let eth_config = try_load_config!(self.secrets.l1);
+        let query_eth_client_layer = QueryEthClientLayer::new(
+            genesis.settlement_layer_id(),
+            eth_config.l1_rpc_url,
+            self.configs
+                .eth
+                .as_ref()
+                .and_then(|x| Some(x.gas_adjuster?.settlement_mode))
+                .unwrap_or(SettlementMode::SettlesToL1),
+        );
+        self.node.add_layer(query_eth_client_layer);
+        Ok(self)
+    }
+
+    fn add_storage_initialization_layer(mut self, kind: LayerKind) -> anyhow::Result<Self> {
+        self.node.add_layer(MainNodeInitStrategyLayer {
```

Comment:
The `MainNodeInitStrategyLayer` layer will init the genesis setup by calling L1 eth contracts [here](https://github.com/vianetwork/via-core/blob/8a23bec129a210b4cb65ae06483484bba7d16b1d/core/node/genesis/src/lib.rs#L408)
We are planning to set the L1 contracts to ZERO, so with such setup the logic will revert when init this layer.

I suggest to reimplement `save_set_chain_id_tx` code [logic](https://github.com/vianetwork/via-core/blob/8a23bec129a210b4cb65ae06483484bba7d16b1d/core/node/genesis/src/lib.rs#L408) and create a new `via_genesis` file [here](https://github.com/vianetwork/via-core/blob/8a23bec129a210b4cb65ae06483484bba7d16b1d/core/node/node_storage_init/src/main_node/genesis.rs#L19), then reimplemnt this [layer](https://github.com/vianetwork/via-core/blob/8a23bec129a210b4cb65ae06483484bba7d16b1d/core/node/node_framework/src/implementations/layers/node_storage_init/main_node_strategy.rs#L18) to use the vai_genesis and remove the `eth_interface: EthInterfaceResource` from inputs.

wdyt?

Reply:
so I disabled the `ecosystem_contracts` [here](https://github.com/vianetwork/via-core/blob/feat/via-genesis/core/bin/via_server/src/main.rs#L86)

I think it good for now since we will have to deal with other logic which uses L1 smart contracts in next `devnet2`